test/cppsuite/test_harness/runtime_monitor.h

```diff
@@ -45,67 +45,62 @@ class configuration;
 
 namespace test_harness {
 
-class runtime_statistic {
+class statistics {
     public:
-    explicit runtime_statistic(configuration *config);
-    virtual ~runtime_statistic() = default;
+    statistics() = default;
+    explicit statistics(configuration *config, const std::string &stat_name, int stat_field);
+    virtual ~statistics() = default;
 
-    /* Check that the given statistic is within bounds. */
-    virtual void check(scoped_cursor &cursor) = 0;
+    /* Check that the statistics are within bounds. */
+    virtual void check(scoped_cursor &cursor);
 
-    bool enabled() const;
+    /* Retrieve the value associated to the stat in a string format. */
+    virtual std::string get_value_str(scoped_cursor &cursor);
+
+    /* Getters. */
+    int get_field() const;
+    int64_t get_max() const;
+    int64_t get_min() const;
+    const std::string &get_name() const;
+    bool get_runtime();
+    bool get_postrun();
 
     protected:
-    bool _enabled = false;
+    int field;
+    int64_t max;
+    int64_t min;
+    std::string name;
+    bool postrun;
+    bool runtime;
 };
 
-class cache_limit_statistic : public runtime_statistic {
+class cache_limit_statistic : public statistics {
     public:
-    explicit cache_limit_statistic(configuration *config);
+    explicit cache_limit_statistic(configuration *config, const std::string &name);
     virtual ~cache_limit_statistic() = default;
 
     void check(scoped_cursor &cursor) override final;
+    std::string get_value_str(scoped_cursor &cursor) override final;
 
     private:
-    int64_t _limit;
+    double get_cache_value(scoped_cursor &cursor);
 };
 
-class db_size_statistic : public runtime_statistic {
+class db_size_statistic : public statistics {
     public:
-    explicit db_size_statistic(configuration *config, database &database);
+    explicit db_size_statistic(configuration *config, const std::string &name, database &database);
     virtual ~db_size_statistic() = default;
 
-    /* Don't need the stat cursor for this. */
+    /* Don't need the stat cursor for these. */
     void check(scoped_cursor &) override final;
```

Comment:
The approach here is that I have defined things to meet today's requirements. In the future, if we ever inherit from that class and need to redefine the function, I would expect the person in charge to remove the keyword `final`.

Reply:
It's just my opinion, and we can leave it as is, but the person in the future would see `final` and think that that is the design and there was a reason for it, and they will probably copy the whole content of the method in their subclass rather than change the base class or find some other ways around it. Adding final is a design decision that says don't override it.

Reply:
Thanks @AlexeyAnisimovAU, I get your point. @jeremythorp, any thoughts?

Reply:
After thinking more about it, I don't think we want to inherit from that class since it is for a specific stat. Each stat that needs to have a specific implementation of the `check` function should be having its own class. I cannot think of any stat where it would be interesting to inherit from `db_size_statistic`.